applications/StructuralMechanicsApplication/custom_elements/spring_damper_element.cpp

```diff
@@ -0,0 +1,566 @@
+
+template<std::size_t TDim>
+void SpringDamperElement<TDim>::CalculateLeftHandSide( MatrixType& rLeftHandSideMatrix, const ProcessInfo& rCurrentProcessInfo )
+{
+    // Resizing the LHS
+    std::size_t system_size = msElementSize;
+
+    if ( rLeftHandSideMatrix.size1() != system_size ) {
+        rLeftHandSideMatrix.resize( system_size, system_size, false );
+    }
+
+    noalias( rLeftHandSideMatrix ) = ZeroMatrix( system_size, system_size ); //resetting LHS
+
+    // elemental_stiffness: kx, ky, kz, cpx, cpy, cpz
+    array_1d<double, msLocalSize > elemental_stiffness = ZeroVector( msLocalSize );
+    array_1d<double, 3> nodal_stiffness = ZeroVector(3);
+    array_1d<double, 3> nodal_rot_stiffness = ZeroVector(3);
+
+    if (this->Has(NODAL_DISPLACEMENT_STIFFNESS))
+    {
+        nodal_stiffness = this->GetValue(NODAL_DISPLACEMENT_STIFFNESS);
+    }
+    if (this->Has(NODAL_ROTATIONAL_STIFFNESS))
+    {
+        nodal_rot_stiffness = this->GetValue(NODAL_ROTATIONAL_STIFFNESS);
+    }
+
+    if constexpr (TDim == 2) {
+
+        elemental_stiffness[0] = nodal_stiffness[0];
+        elemental_stiffness[1] = nodal_stiffness[1];
+        elemental_stiffness[2] = nodal_rot_stiffness[2];
+
+    }
+    else if constexpr (TDim == 3) {
+        elemental_stiffness[0] = nodal_stiffness[0];
+        elemental_stiffness[1] = nodal_stiffness[1];
+        elemental_stiffness[2] = nodal_stiffness[2];
+
+        elemental_stiffness[3] = nodal_rot_stiffness[0];
+        elemental_stiffness[4] = nodal_rot_stiffness[1];
+        elemental_stiffness[5] = nodal_rot_stiffness[2];
+    }
+
+    for ( std::size_t i = 0; i < msLocalSize; ++i ) {
+        rLeftHandSideMatrix(i    ,i    ) += elemental_stiffness[i];
+        rLeftHandSideMatrix(i + msLocalSize,i + msLocalSize) += elemental_stiffness[i];
+        rLeftHandSideMatrix(i    ,i + msLocalSize) -= elemental_stiffness[i];
+        rLeftHandSideMatrix(i + msLocalSize,i    ) -= elemental_stiffness[i];
+    }
+
+}
+
+//************************************************************************************
+//************************************************************************************
+
+template<std::size_t TDim>
+void SpringDamperElement<TDim>::CalculateMassMatrix( MatrixType& rMassMatrix, const ProcessInfo& rCurrentProcessInfo )
+{
+    KRATOS_TRY
+
+    //this is a massless element
+    std::size_t system_size = msElementSize;
+
+    if ( rMassMatrix.size1() != system_size )
+    {
+        rMassMatrix.resize( system_size, system_size, false );
+    }
+
+    rMassMatrix = ZeroMatrix( system_size, system_size );
+
+    KRATOS_CATCH( "" );
+}
+
+//************************************************************************************
+//************************************************************************************
+
+template<std::size_t TDim>
+void SpringDamperElement<TDim>::CalculateDampingMatrix( MatrixType& rDampingMatrix, const ProcessInfo& rCurrentProcessInfo )
```

Comment:
same, please add a `ConstCalculateDampingMatrix`, which is called from `CalculateDampingMatrix`